### ISSUE TITLE
PHP notice 'Undefined variable: user' pops up at a few code locations

### ISSUE THREAD

**Issue by rwetzlmayr** [SocialEngine/phpv4-issues#325]
A few copy/paste occurrences of an inexistent variable `$user` instead of the proper `$viewer`keep filling up our main.log.

Here's a diff to fix that:

```diff
--- a/web/application/modules/Forum/Form/Post/Create.php
+++ b/web/application/modules/Forum/Form/Post/Create.php
@@ -45,7 +45,7 @@ class Forum_Form_Post_Create extends Engine_Form
     if( $allowHtml || $allowBbcode ) {
       $upload_url = "";
     
-      if(Engine_Api::_()->authorization()->isAllowed('album', $user, 'create')){
+      if(Engine_Api::_()->authorization()->isAllowed('album', $viewer, 'create')){
         $upload_url = Zend_Controller_Front::getInstance()->getRouter()->assemble(array('action'=>'upload-photo'),'forum_photo',true);
 
       }
@@ -123,4 +123,4 @@ class Forum_Form_Post_Create extends Engine_Form
     $button_group->addDecorator('DivDivDivWrapper');
   }
 
-}
\ No newline at end of file
+}

--- a/web/application/modules/Forum/Form/Post/Quick.php
+++ b/web/application/modules/Forum/Form/Post/Quick.php
@@ -45,7 +45,7 @@ class Forum_Form_Post_Quick extends Engine_Form
 
       $upload_url = "";
     
-      if(Engine_Api::_()->authorization()->isAllowed('album', $user, 'create')){
+      if(Engine_Api::_()->authorization()->isAllowed('album', $viewer, 'create')){
         $upload_url = Zend_Controller_Front::getInstance()->getRouter()->assemble(array('action'=>'upload-photo'),'forum_photo',true);
 
       }
@@ -112,4 +112,4 @@ class Forum_Form_Post_Quick extends Engine_Form
       'type' => 'submit',
     ));
   }
-}
\ No newline at end of file
+}

--- a/web/application/modules/Forum/Form/Topic/Create.php
+++ b/web/application/modules/Forum/Form/Topic/Create.php
@@ -53,7 +53,7 @@ class Forum_Form_Topic_Create extends Engine_Form
     if( $allowHtml || $allowBbcode ) {
       $upload_url = "";
     
-      if(Engine_Api::_()->authorization()->isAllowed('album', $user, 'create')){
+      if(Engine_Api::_()->authorization()->isAllowed('album', $viewer, 'create')){
         $upload_url = Zend_Controller_Front::getInstance()->getRouter()->assemble(array('action'=>'upload-photo'),'forum_photo',true);
 
       }
@@ -125,4 +125,4 @@ class Forum_Form_Topic_Create extends Engine_Form
     $button_group = $this->getDisplayGroup('buttons');
     $button_group->addDecorator('DivDivDivWrapper');
   }
-}
\ No newline at end of file
+}

--- a/web/application/modules/Group/Form/Topic/Create.php
+++ b/web/application/modules/Group/Form/Topic/Create.php
@@ -54,7 +54,7 @@ class Group_Form_Topic_Create extends Engine_Form
     if ( $allowHtml || $allowBbcode ) {
       $upload_url = "";
     
-      if(Engine_Api::_()->authorization()->isAllowed('album', $user, 'create')){
+      if(Engine_Api::_()->authorization()->isAllowed('album', $viewer, 'create')){
         $upload_url = Zend_Controller_Front::getInstance()->getRouter()->assemble(array('controller'=>'index','action'=>'upload-photo'),'group_extended',true);
 
       }
@@ -129,4 +129,4 @@ class Group_Form_Topic_Create extends Engine_Form
 
     $this->addDisplayGroup(array('submit', 'cancel'), 'buttons');
   }
-}
\ No newline at end of file
+}

--- a/web/application/modules/Messages/Form/Compose.php
+++ b/web/application/modules/Messages/Form/Compose.php
@@ -66,7 +66,7 @@ class Messages_Form_Compose extends Engine_Form
           'html' => true,
       );
 
-      if(Engine_Api::_()->authorization()->isAllowed('album', $user, 'create')){
+      if(Engine_Api::_()->authorization()->isAllowed('album', null, 'create')){
         $upload_url = Zend_Controller_Front::getInstance()->getRouter()->assemble(array('action' => 'upload-photo'), 'messages_general', true);
       }
 
@@ -123,4 +123,4 @@ class Messages_Form_Compose extends Engine_Form
       'ignore' => true
     ));
   }
-}
\ No newline at end of file
+}
```

**Comment by smazurov** [comment]
another great patch, thanks! We'll verify and add it in the next release.

**Comment by smazurov** [comment]
Fixed in [4.8.12](http://blog.socialengine.com/2016/07/07/socialengine-php-4-8-12-is-here/)
